flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/SchemaOperator.java

```diff
@@ -270,10 +276,18 @@ private RecordData regenerateRecordData(
     private List<TableId> getRoutedTables(TableId originalTableId) {
         return routes.stream()
                 .filter(route -> route.f0.isMatch(originalTableId))
-                .map(route -> route.f1)
+                .map(route -> resolveReplacement(originalTableId, route))
                 .collect(Collectors.toList());
     }
 
+    private TableId resolveReplacement(
+            TableId originalTable, Tuple3<Selectors, String, String> route) {
+        if (route.f2 != null) {
+            return TableId.parse(route.f1.replace(route.f2, originalTable.getTableName()));
```

Comment:
Can we use Map<TableId, TableId> to reduce this replace and parse operation?

Reply:
Done, used `LoadingCache` to store table mapping relations. As for previously discussed `hashCode` caching optimization, I'll open another PR to address it.